shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.mm

```diff
@@ -2660,6 +2660,41 @@ - (void)testFlutterTokenizerCanParseLines {
   XCTAssertEqual(range.range.length, 20u);
 }
 
+- (void)testFlutterTokenizerEndOfDocumentLineRangeQuery {
+  FlutterTextInputView* inputView = [[FlutterTextInputView alloc] initWithOwner:textInputPlugin];
+  id<UITextInputTokenizer> tokenizer = [inputView tokenizer];
+
+  NSString* text = @"Random text\nwith 2 lines";
+  [inputView insertText:text];
+
+  FlutterTextPosition* endOfDocumentWithBackwardAffinity =
+      [FlutterTextPosition positionWithIndex:text.length affinity:UITextStorageDirectionBackward];
+
+  FlutterTextRange* range =
+      (FlutterTextRange*)[tokenizer rangeEnclosingPosition:endOfDocumentWithBackwardAffinity
+                                           withGranularity:UITextGranularityLine
+                                               inDirection:UITextLayoutDirectionRight];
+
+  if (@available(iOS 17.0, *)) {
+    XCTAssertNil(range);
+  } else {
+    XCTAssertEqual(range.range.location, 12u);
+    XCTAssertEqual(range.range.length, 12u);
+  }
+
+  // End of document with forward affinity should still be considered as part of the document.
```

Comment:
This seems like two distinct tests, so should be written as such. Unit tests should generally only test one behavior each unless there's a compelling reason to combine behaviors.